addons/web/static/src/views/sample_server.js

```diff
@@ -673,11 +679,13 @@ export class SampleServer {
         // update count and aggregates for each group
         const fullGroupBy = params.groupBy[0];
         const groupBy = fullGroupBy.split(":")[0];
+        const groupByField = this.data[params.model].fields[groupBy];
         const records = this.data[params.model].records;
+        const fn = getProcessFn(groupByField.type);
```

Comment:
naming could be improved. Also not sure about having a function that returns the "conversion function". Maybe simply call a function that takes a type and a value and returns the server value.